web/Web/web/static/js/document-viewer.js

```diff
@@ -555,13 +604,20 @@ docView.prototype = {
         });
       }
       // link on click
-      div_elm.on("click", function(){viewPreviouslyJudgedDocument($(this).data("doc-id").toString())});
-
-      // add to top of the list view/stack
-      $(options.previouslyReviewedListSelector).prepend(div_elm);
-      parent.previouslyJudgedDocsStack.push(docid);
-
+      div_elm.on("click", function () {viewPreviouslyJudgedDocument($(this).data("doc-id").toString())});
 
+      if (!options.reviewMode || stack_index === -1) {
+        // add to top of the list view/stack
+        $(options.previouslyReviewedListSelector).prepend(div_elm);
+        parent.previouslyJudgedDocsStack.push(docid);
+      } else {
+        // update indicator
+        $("." + options.prevReviewedDocumentItemClass).each(function () {
```

Comment:
It seems a inefficient to go over all the prev reviewed documents in the list to change one docid. Is there a better way to do this?

Reply:
Got rid of the loop